Project-Hotpot/Project-Hotpot/Parse-Backend/PFPlaylist.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  PFPlaylist.swift
+//  Project-Hotpot
+//
+//  Created by Harini Sundaram on 7/18/22.
+//
+
+import Foundation
+import Parse
+import AFNetworking
+
+class PFPlaylist: PFObject, PFSubclassing {
+    @NSManaged var user : PFUser
+    @NSManaged var name : String?
+    @NSManaged var songArray : [PFSong]?
+    
+    static func parseClassName() -> String {
+        return "Playlist"
+    }
+    
+    class func createPlaylist(name:String, completion: (Bool) -> Void){
+        let newPlaylist = PFPlaylist()
+        if(PFUser.current() == nil){
+            NSLog("current user is nil")
+            completion(false)
+        }
+        newPlaylist.user = PFUser.current()!
+        newPlaylist.name = name
+        newPlaylist.songArray = []
+        
+        do {
+            //save synchronously, since creation will most likely be followed by edits to the playlist
+            //TODO: Feedback on synchronous creation or async? If synchronous, is completion necessary?
+            try newPlaylist.save()
```

Comment:
It does look (from the docs) like this is a synchronous network call.
They likely included this to allow for more straightforward code that's _already on_ a background thread. But if saving from the main thread, you'll want to stick with `saveInBackground` for the reasons Harsh listed above.
It's a common pattern to 1) show a loading indicator / disable some UI, 2) fire off a network call, and then, 3) stop the loading indicator / re-enable UI after the call finishes. This will allow you to gracefully block the user instead of freezing the UI (apps that clearly communicate their state can still _feel_ performant, even if they need to take moments here and there to pause and talk with the server).